docs/architecture/log_levels.md

```diff
@@ -0,0 +1,122 @@
+# Log levels
+
+Using "actor" here to describe a repeated or ongoing routine in launcher. It includes all rungroup actors, plus launcher itself, plus some other one-offs.
+
+## What we need logs and log levels for
+
+When troubleshooting, we want to be able to distinguish between and easily find the following categories of logs:
+
+* Positive confirmation that launcher is operating as expected (routine actions are succeeding)
+* Contextual status information (how launcher is performing at a given time)
+* Error information (any time launcher tries to perform an action but cannot)
+* Critical error information (any time an actor exits and must be restarted due to an error)
+
+We want logs to be stored by level in different locations:
+
+* Stored locally and collected via flare, to troubleshoot individual devices
+* Shipped to Cloud Log, to troubleshoot devices across a tenant or multiple tenants
+* Shipped to Cloud Log and Error Reporting, to identify error trends that might indicate launcher bugs
+
+## Proposal
+
+We should use the same five levels we already have -- we don't need to add additional complexity by adding more, especially since we don't have more categories of logs to cover. We should use the following definitions to level our logs:
+
+* LevelDebug: Routine information, such as ongoing status or performance
+* LevelInfo: Normal but significant events, such as start up or shut down
+* LevelWarn: Error occurred that is temporary or that does not prevent the action from continuing
+* LevelError: Error occurred that prevents action from continuing
+* LevelReportedError: Error occurred that results in actor shutdown
```

Comment:
```suggestion
* LevelReportedError: Error occurred that results in actor shutdown, or error occurred that prevents very important action from continuing
```

Reply:
https://github.com/kolide/launcher/pull/2305/files has all of the `LevelError` messages that I upgraded to `LevelReportedError` -- I reviewed all the `LevelError` logs in launcher and those were the ones I decided to upgrade. I will put actual examples into `Example categorized logs in launcher` section as well.